core/src/main/java/org/mule/execution/AsyncResponseFlowProcessingPhase.java

```diff
@@ -98,6 +98,18 @@ private void fireNotification(MuleEvent event, int action)
     {
         try
         {
+            if (event == null)
```

Comment:
We are trying to retrieve the last available event here, because filters set nullify the event and the notification needs an event to being fired.